src/main/java/org/springframework/data/domain/ExampleSpec.java

```diff
@@ -0,0 +1,797 @@
+	/**
+	 * Create a new {@link ExampleSpec} based on the current {@link ExampleSpec} and ignore case sensitivity for the
+	 * {@code propertyPaths}.
+	 *
+	 * @param propertyPaths must not be {@literal null} and not empty.
+	 * @return
+	 */
+	public ExampleSpec<T> withIgnoreCase(String... propertyPaths) {
+
+		Assert.notEmpty(propertyPaths, "PropertyPaths must not be empty!");
+		Assert.noNullElements(propertyPaths, "PropertyPaths must not contain null elements!");
+
+		PropertySpecifiers propertySpecifiers = new PropertySpecifiers(this.propertySpecifiers);
+
+		for (String propertyPath : propertyPaths) {
+			PropertySpecifier propertySpecifier = createOrClonePropertySpecifier(propertyPath, propertySpecifiers);
+			propertySpecifier.ignoreCase = true;
+			propertySpecifiers.add(propertySpecifier);
+		}
+
+		return new ExampleSpec<T>(type, nullHandler, defaultStringMatcher, propertySpecifiers, ignoredPaths,
+				defaultIgnoreCase);
+	}
+
+	private PropertySpecifier createOrClonePropertySpecifier(String propertyPath, PropertySpecifiers propertySpecifiers) {
+		PropertySpecifier propertySpecifier;
+
+		if (propertySpecifiers.hasSpecifierForPath(propertyPath)) {
+			propertySpecifier = new PropertySpecifier(propertyPath);
+			PropertySpecifier existing = propertySpecifiers.getForPath(propertyPath);
+			propertySpecifier.ignoreCase = existing.ignoreCase;
+			propertySpecifier.stringMatcher = existing.stringMatcher;
+		} else {
+			propertySpecifier = new PropertySpecifier(propertyPath);
+		}
+		return propertySpecifier;
+	}
+
+	/**
+	 * Create a new {@link ExampleSpec} based on the current {@link ExampleSpec} and set treatment of {@literal null}
+	 * values to {@link NullHandler#INCLUDE}.
+	 *
+	 * @return
+	 */
+	public ExampleSpec<T> withIncludeNullValues() {
+		return new ExampleSpec<T>(type, NullHandler.INCLUDE, defaultStringMatcher, propertySpecifiers, ignoredPaths,
+				defaultIgnoreCase);
+	}
+
+	/**
+	 * Create a new {@link ExampleSpec} based on the current {@link ExampleSpec} and set treatment of {@literal null}
+	 * values to {@link NullHandler#IGNORE}.
+	 *
+	 * @return
+	 */
+	public ExampleSpec<T> withIgnoreNullValues() {
+		return new ExampleSpec<T>(type, NullHandler.IGNORE, defaultStringMatcher, propertySpecifiers, ignoredPaths,
+				defaultIgnoreCase);
+	}
+
+	/**
+	 * Create a new {@link ExampleSpec} based on the current {@link ExampleSpec} and set treatment of {@literal null}
+	 * values to {@code nullHandler}.
+	 *
+	 * @param nullHandler must not be {@literal null}.
+	 * @return
+	 */
+	public ExampleSpec<T> withNullHandler(NullHandler nullHandler) {
+
+		Assert.notNull(nullHandler, "NullHandler must not be null!");
+		return new ExampleSpec<T>(type, nullHandler, defaultStringMatcher, propertySpecifiers, ignoredPaths,
+				defaultIgnoreCase);
+	}
+
+	/**
+	 * Create a new {@link Example} containing the {@code probe}.
+	 *
+	 * @param probe must not be {@literal null}.
+	 * @return
+	 */
+	public Example<T> createExample(T probe) {
```

Comment:
Can we avoid this method being exposed or maybe even remove it at all? We're creating ambiguity here which method should be preferred to be used (`Example.of(…)` or this one here).